bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -0,0 +1,548 @@
+        return "openhab";
+    }
+
+    @Override
+    public Object execute(Session session, List<Object> argList) throws Exception {
+        String[] args = argList.stream().map(Object::toString).toArray(String[]::new);
+        PrintStream out = Process.Utils.current().out();
+        final Console console = new JRubyConsole(getScope(), out, session);
+
+        if (args.length > 0) {
+            String command = args[0];
+            switch (command) {
+                case INFO:
+                    info(console);
+                    break;
+                case SHELL:
+                    shell(console);
+                    break;
+                case GEM:
+                    gem(console, Arrays.stream(args).skip(1).toList());
+                    break;
+                case UPDATE:
+                    updateGems(console);
+                    break;
+                case PRUNE:
+                    if (args.length > 1) {
+                        if ("-f".equals(args[1]) || "--force".equals(args[1])) {
+                            cleanupOtherGemHomes(console, true);
+                        } else {
+                            console.println("Use -f or --force to skip confirmation.");
+                        }
+                    } else {
+                        cleanupOtherGemHomes(console, false);
+                    }
+                    break;
+                case UNLOAD:
+                    unloadEngine();
+                    break;
+                case "--help":
+                    printUsage(console);
+                    break;
+                default:
+                    console.println("Unknown command '" + command + "'");
+                    printUsage(console);
+                    break;
+            }
+        } else {
+            printUsage(console);
+        }
+        return null;
+    }
+
+    private void info(Console console) {
+        final String PRINT_VERSION_NUMBERS = """
+                library_version = Module.const_defined?(:OpenHAB) && OpenHAB::DSL::VERSION
+
+                puts "JRuby #{JRUBY_VERSION}"
+                puts "JRuby Scripting Library #{library_version || 'is not installed'}"
+                puts "GEM_HOME: #{ENV['GEM_HOME']}"
+                puts "RUBYLIB: #{ENV['RUBYLIB']}"
+                    """;
+
+        executeWithJRuby(console, engine -> engine.eval(PRINT_VERSION_NUMBERS));
+        console.println("Script path: " + scriptFileWatcher.getWatchPath());
+        console.println("");
+        console.println("JRuby Scripting Add-on Configuration:");
+        console.println("=====================================");
+        jRubyScriptEngineFactory.getConfiguration().getConfigurations().forEach((key, value) -> {
+            console.println(key + ": " + value);
+        });
+    }
+
+    private void shell(Console console) {
+        // We're doing the REPL inside Ruby to preserve local variables between loops
+        final String RUBY_REPL = """
```

Comment:
this is a lot of Ruby to embed in the Java addon. I'd much prefer if we can keep this in the helper library, and simply have an (advanced) configuration for how to start the console (which defaults to something like `require "openhab/dsl/console"`). that allows us iterate more quickly on how it works. Cause ideally I'd love to get this to _actually_ be backed by `irb` (which would both eliminate the odd Alt-Enter shortcut for multi-line blocks, improve the printing of results, and probably other improvements on this basic implementation).

Reply:
excellent idea! We'll get syntax highlighting and auto formatting that way too.